lms/views/lti/deeplinking.py

```diff
@@ -58,11 +68,83 @@ def content_item_selection(context, request):
     request.find_service(name="lti_h").sync([context.h_group], request.params)
 
     context.js_config.enable_content_item_selection_mode(
-        form_action=request.params["content_item_return_url"],
+        form_action=request.parsed_params["content_item_return_url"],
         form_fields={
             "lti_message_type": "ContentItemSelection",
-            "lti_version": request.params["lti_version"],
+            "lti_version": request.parsed_params["lti_version"],
         },
     )
 
     return {}
+
+
+class DeeplinkingFieldsRequestSchema(JSONPyramidRequestSchema):
+    deep_linking_settings = fields.Dict(required=False)
+    content_item_return_url = fields.Str(required=True)
+    content = fields.Dict(required=True)
+
+    extra_params = fields.Dict(required=False)
+
+
+@view_defaults(
+    request_method="POST", renderer="json", schema=DeeplinkingFieldsRequestSchema
+)
+class DeeplinkingFieldsViews:
+    """
+    Views that return the required form fields to complete a deeplinking request to the LMS.
+
+    After the user picks a document to be annotated the frontend will call these views
+    to get all the necessary fields to configure the assignment submitting them in a form to the LMS.
+    """
+
+    def __init__(self, request):
+        self.request = request
+
+        self.application_instance = self.request.find_service(
+            name="application_instance"
+        ).get_current()
+
+    @view_config(route_name="lti.deeplinking.formfields")
+    def formfields_v13(self):
```

Comment:
>  the front-end uses this to convert our standard file-picker output back into a set of data it can pass on to LMS to complete the deeplinking request?

Yes, exactly

I went for `file_picker_to_formfields_v13` (we are already on deeplinking.py).